clients/http-client-common/src/main/java/org/apache/servicecomb/http/client/common/AbstractAddressManager.java

```diff
@@ -148,10 +151,21 @@ public String formatUrl(String url, boolean absoluteUrl, String address) {
   }
 
   public String address() {
+    String address = "";
     if (!addressAutoRefreshed) {
-      return getDefaultAddress();
+      address = getDefaultAddress();
+    } else {
+      address = getAvailableZoneAddress();
+    }
+    if (telnetTest(address)) {
+      return address;
     } else {
-      return getAvailableZoneAddress();
+      if (addressRetry.getAndIncrement() >= 3) {
```

Comment:
This is not correct. And it is better to add a tests case for choose address.

Reply:
fixed